core/services/workflows/engine.go

```diff
@@ -58,26 +46,34 @@ func (e *Engine) init(ctx context.Context) {
 	retrySec := 5
 	ticker := time.NewTicker(time.Duration(retrySec) * time.Second)
 	defer ticker.Stop()
+
+	// Note: in our hardcode workflow, there is only one trigger,
+	// and one consensus step.
+	trigger := e.workflow.Triggers[0]
+	consensus := e.workflow.Consensus[0]
+
 	var err error
 LOOP:
 	for {
 		select {
 		case <-ctx.Done():
 			return
```

Comment:
No, this is a possible outcome; the ctx.Done() is because we've decided to shut down.

Reply:
hmm. won't this loop block forever if there are errors and the ctx doesn't have a timeout? wouldn't the ctx.Err be the right way to right way to report that you had to quit because it was taking too long?